jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -972,6 +972,30 @@ public synchronized void checkpointJVM() {
 				J9InternalCheckpointHookAPI.registerPostRestoreHook(HookMode.SINGLE_THREAD_MODE, RESTORE_CLEAR_INETADDRESS_CACHE_PRIORITY, "Clear InetAddress cache on restore", InternalCRIUSupport::clearInetAddressCache); //$NON-NLS-1$
 				J9InternalCheckpointHookAPI.registerPostRestoreHook(HookMode.SINGLE_THREAD_MODE, RESTORE_ENVIRONMENT_VARIABLES_PRIORITY, "Restore system properties", InternalCRIUSupport::setRestoreJavaProperties); //$NON-NLS-1$
 
+				/* Add option overrides */
+				if (System.getProperty("openj9.internal.criu.unprivilegedMode") != null) { //$NON-NLS-1$
+					setUnprivileged(Boolean.getBoolean("openj9.internal.criu.unprivilegedMode")); //$NON-NLS-1$
+				}
+
+				if (System.getProperty("openj9.internal.criu.tcpEstablished") != null) { //$NON-NLS-1$
+					setTCPEstablished(Boolean.getBoolean("openj9.internal.criu.tcpEstablished")); //$NON-NLS-1$
+				}
+
+				String ghostFileLimitOpt = System.getProperty("openj9.internal.criu.ghostFileLimit"); //$NON-NLS-1$
+				if (ghostFileLimitOpt != null) {
+					setGhostFileLimit(Long.parseLong(ghostFileLimitOpt));
+				}
+
+				String logLevelOpt =  System.getProperty("openj9.internal.criu.logLevel"); //$NON-NLS-1$
+				if (logLevelOpt != null) {
+					setLogLevel(Integer.parseInt(logLevelOpt));
```

Comment:
Similarly, `NumberFormatException()` needs to be caught, and maybe a default `logLevel`?